loyalty_mass_mailing/models/mailing_mailing.py

```diff
@@ -1,17 +1,12 @@
 # Copyright 2021 Tecnativa - Víctor Martínez
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
-from odoo import api, fields, models
+from odoo import fields, models
 
 
 class MailingMailing(models.Model):
     _inherit = "mailing.mailing"
 
     program_id = fields.Many2one(
-        comodel_name="coupon.program", string="Program", ondelete="cascade"
+        comodel_name="loyalty.program", string="Program", ondelete="cascade"
     )
-
-    @api.onchange("program_id")
-    def onchange_program_id(self):
-        if self.program_id:
-            self.mailing_domain = self.program_id.rule_partners_domain
```

Comment:
Maybe we'll be adding a module to recover the partner's domain filter. I guess it was dropped for performance, but some customers rely on this. And then this module will depend on that one (just commenting aloud)

Reply:
Here it is. Let's add it as a dependency to this one: https://github.com/OCA/sale-promotion/pull/163

Reply:
ok, but I think will move this functionality to loyalty.rules records, what you think about this?, because is currently in loyalty.program

Reply:
Yes, the models relations have changed, so you have to take the domain from there

Reply:
@chienandalu I maked this changes, but I have a question: what should be with translate files? will generate again?

Reply:
You should at least rename the references to the old module name